src/data/polynomial/degree/erase_leading.lean

```diff
@@ -0,0 +1,166 @@
+/-
+Copyright (c) 2020 Damiano Testa. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Damiano Testa
+-/
+
+import data.polynomial.degree.basic
+import data.polynomial.degree.to_basic
+import data.polynomial.degree.to_trailing_degree
+import data.polynomial.degree.trailing_degree
+
+open polynomial finsupp finset
+
+namespace polynomial
+
+variables {R : Type*} [semiring R] {f : polynomial R}
+
+namespace erase_leading
```

Comment:
```suggestion
namespace erase_lead
```
I think it'd be better if the namespace and the filename match the name of the function. I don't mind if you settle on `erase_lead` or `erase_leading`, though.

Reply:
I matched the namespace with the name of the function, but I got a lint error saying precisely that this should not happen.  Should I ignore the error message?

Reply:
Ah. Right now, the function itself is in the `erase_leading` namespace, which means the function itself is called `polynomial.erase_leading.erase_lead`. That repetition is unnecessary, and the linter's trying to keep you from naming it something even more repetitive, like `polynomial.erase_lead.erase_lead`. You should probably pull `erase_lead` out of the namespace altogether, so that it's just called `polynomial.erase_lead`, and then you can call the namespace `erase_lead`.